File: app/src/main/java/com/digicoffer/lauditor/Meetings/Models/CalendarItem.java
package com.digicoffer.lauditor.Meetings.Models;

import com.digicoffer.lauditor.Appointments.Models.AppointmentModel;

public class CalendarItem {
    private Events_Do event;
    private AppointmentModel appointment;
    private ItemType type;

    public enum ItemType {
        EVENT, APPOINTMENT
    }

    // Constructor for Event
    public CalendarItem(Events_Do event) {
        this.event = event;
        this.type = ItemType.EVENT;
    }

    // Constructor for Appointment
    public CalendarItem(AppointmentModel appointment) {
        this.appointment = appointment;
        this.type = ItemType.APPOINTMENT;
    }

    public ItemType getType() {
        return type;
    }

    public Events_Do getEvent() {
        return event;
    }

    public AppointmentModel getAppointment() {
        return appointment;
    }

    public boolean isEvent() {
        return type == ItemType.EVENT;
    }

    public boolean isAppointment() {
        return type == ItemType.APPOINTMENT;
    }
}

File: app/src/main/java/com/digicoffer/lauditor/Meetings/Models/CalendarItem.kt
package com.digicoffer.lauditor.Meetings.Models

import com.digicoffer.lauditor.Appointments.Models.AppointmentModel

class CalendarItem {
    var event: Events_Do? = null
        private set
    var appointment: AppointmentModel? = null
        private set
    var type: ItemType? = null
        private set

    enum class ItemType {
        EVENT, APPOINTMENT
    }

    constructor(event: Events_Do) {
        this.event = event
        this.type = ItemType.EVENT
    }

    constructor(appointment: AppointmentModel) {
        this.appointment = appointment
        this.type = ItemType.APPOINTMENT
    }

    fun isEvent(): Boolean {
        return type == ItemType.EVENT
    }

    fun isAppointment(): Boolean {
        return type == ItemType.APPOINTMENT
    }
}
